Core/include/Acts/Utilities/DynamicArray.hpp

```diff
@@ -0,0 +1,216 @@
+// This file is part of the Acts project.
```

Comment:
for the record, I am not convinced we need to switch this out 😇

Reply:
@junggjo9  :)

Reply:
It solves the problem for any kind of dimensional array. Applying occam's razor, the more general the better. We do

Reply:
That is the **opposite** of occam's razor! 
We don't need any kind of dimensionality, and the entire remaining codebase is hardcoded to be 2D

Reply:
for the hough it is true, but I do not know if it would be nice to have it in acts for also other purposes?